### ISSUE TITLE
periph/spi: undefined behavior for length > MAX_INT

### ISSUE THREAD

**Issue by LudwigKnuepfer** [RIOT-OS/RIOT#1769]
The documentation for `spi_transfer_regs` (and others) does not specify what happens if the `length` parameter exceeds the maximum value the return type can take.

For reference:

``` c
/**
 * @brief Transfer a number of bytes from/to a given register address
 *
 * This function is a shortcut function for easier handling of register based SPI devices. As
 * many SPI devices use a register based addressing scheme, this function is a convenient short-
 * cut for interfacing with such devices.
 *
 * @param[in] dev       SPI device to use
 * @param[in] reg       Register address to transfer data to/from
 * @param[in] out       Byte array to send data from, set NULL if only receiving
 * @param[out] in       Byte buffer to read into, set NULL if only sending
 * @param[in] length    Number of bytes to transfer
 *
 * @return              Number of bytes that were transfered
 * @return              -1 on error
 */
int spi_transfer_regs(spi_t dev, uint8_t reg, char *out, char *in, unsigned int length);
```

I did not check whether this type of documentation error exists for other periph interfaces as well. It should be done.

**Comment by OlegHahm** [comment]
Digging in the past: is this still unfixed?

**Comment by PeterKietzmann** [comment]
It is still unfixed. Not sure what makes most sense here. The problem is that you can theoretically send **`unsigned`**` int length` bytes and will return **`signed`**` int length` as the number of transferred bytes. IMO the easiest solutions might be:
i) change one of the types. `return` should stay a `signed` value and increasing it does not make too much sense to me. So maybe reduce `length` type to a 16 bit signed value?
ii) add a comment to the documentation that this case is not handled (does not look too nice but prevents an API change)

**Comment by haukepetersen** [comment]
I think we can close this issue. It does still exist so far, but with #3216 this will be obsolete... If we really feel like addressing this in the mean time (which I think is not worth the effort), lets add a comment.

**Comment by OlegHahm** [comment]
How about closing this issue as soon as #3216 is merged?

**Comment by haukepetersen** [comment]
makes sense

**Comment by haukepetersen** [comment]
#3216 was superseded by #4780, which did not make it into the release, so postponed

**Comment by kaspar030** [comment]
any new insights here?

**Comment by kYc0o** [comment]
#4780 is still being discussed, so this issue will go for the next release.

**Comment by gebart** [comment]
Fixed in #4780. 
 spi_transfer_xx no longer have return values.
